lib/DynamicSegments/RequirementsChecker.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace MailPoet\DynamicSegments;
+
+use MailPoet\DynamicSegments\Filters\Filter;
+use MailPoet\Models\DynamicSegment;
+use MailPoet\WooCommerce\Helper;
+
+class RequirementsChecker {
+
+  /** @var Helper */
+  private $woocommerce_helper;
+
+  function __construct(Helper $woocommerce_helper = null) {
+    if (!$woocommerce_helper) {
+      $woocommerce_helper = new Helper();
+    }
+    $this->woocommerce_helper = $woocommerce_helper;
+  }
+
+  function shouldSkipSegment(DynamicSegment $segment) {
+    foreach ($segment->getFilters() as $filter) {
+      if ($this->shouldSkipFilter($filter)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private function shouldSkipFilter(Filter $filter) {
+    if ($this->woocommerce_helper->isWooCommerceActive()) {
+      return false;
+    }
+    $class = get_class($filter);
+    return strpos($class, 'WooCommerce') !== false;
```

Comment:
This is a bit risky in case of some refactors, etc. Could we make use of https://github.com/mailpoet/mailpoet/blob/master/lib/DynamicSegments/Filters/WooCommerceCategory.php#L10? I know constants are not enforceable by interface but we can either add a `getSegmentType` method to the interface or at least check if the constant exists.